cpp/src/unary/math_ops.cu

```diff
@@ -230,6 +230,59 @@ struct DeviceNot {
   }
 };
 
+template <typename T>
+struct fixed_point_ceil {
+  T n;
+  __device__ T operator()(T data)
+  {
+    T const a = (data / n) * n;                  // result of integer division
+    return a + (data > 0 && a != data ? n : 0);  // add 1 if positive and not round number
+  }
+};
+
+template <typename T>
+struct fixed_point_floor {
+  T n;
+  __device__ T operator()(T data)
+  {
+    T const a = (data / n) * n;                  // result of integer division
+    return a - (data < 0 && a != data ? n : 0);  // subtract 1 if negative and not round number
+  }
+};
+
+template <typename T>
+struct fixed_point_abs {
+  T n;
+  __device__ T operator()(T data) { return std::abs(data); }
+};
+
+template <typename T, template <typename> typename FixedPointFunctor>
+std::unique_ptr<column> unary_op_with(column_view const& input,
+                                      cudaStream_t stream,
+                                      rmm::mr::device_memory_resource* mr)
+{
+  using Type                     = device_storage_type_t<T>;
+  using FixedPointUnaryOpFunctor = FixedPointFunctor<Type>;
+
+  if (input.type().scale() >= 0 &&
```

Comment:
A comment here would be nice. This will be an easy spot to miss when someone adds a unary op that should run with a scale >=0.

Reply:
Yeah, what is this if checking?

Reply:
CEIL and FLOOR are no-ops when scale isn't negative

Reply:
Ah, right.